src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java

```diff
@@ -142,4 +147,71 @@ public boolean isElementVisible(WebElement element) {
   public void openHome(String wikiURL) {
     getUrl(wikiURL);
   }
+
+  /**
+   * Make sure element is ready to be clicked and click on it The separation of this method has
+   * particular reason. It allows global modification of such click usages. This way it is very easy
+   * to control what criteria have to be met in order to click on element
+   *
+   * @param element to be clicked on
+   */
+  protected void waitAndClick(WebElement element) {
+    wait.forElementClickable(element).click();
+  }
+
+  protected void waitAndSendKeys(WebElement element, String keys) {
+    wait.forElementVisible(element);
+    element.sendKeys(keys);
+  }
+
+  protected void verifyTextInElement(WebElement element, String text) {
+    wait.forElementVisible(element);
+    Assertion.assertEquals(element.getText(), text);
+  }
+
+  /**
+   * Verify if element inside element of the provided list has given text. This method assumes list
+   * element is not the final target of verification This method assumes list element is the parent
+   * of target element *
```

Comment:
remove the * at the end of the line.
For me this comment is not very clear. Is it like this list contains of webelements and each element of this list should have the text?